### ISSUE TITLE
Can't mount Extended DSK images

### ISSUE THREAD

**Issue by thejpster** [MiSTer-devel/Amstrad-PCW_MiSTer#4]
I have a bunch of .DSK files I used with my FlashFloppy unit in my PCW9512. I have 180K images, 180K images which are double stepped (to work around an issue on the FlashFloppy with the PCW9512 double stepping the head on a 180K disk), and some 720K images. The Linux `file` command describes them as 'Extended DSK', and a random 180K disk I picked starts with this header:

```
00000000  45 58 54 45 4e 44 45 44  20 43 50 43 20 44 53 4b  |EXTENDED CPC DSK|
00000010  20 46 69 6c 65 0d 0a 44  69 73 6b 2d 49 6e 66 6f  | File..Disk-Info|
00000020  0d 0a 53 41 4d 64 69 73  6b 31 35 30 37 30 38 00  |..SAMdisk150708.|
00000030  28 01 00 00 13 13 13 13  13 13 13 13 13 13 13 13  |(...............|
00000040  13 13 13 13 13 13 13 13  13 13 13 13 13 13 13 13  |................|
00000050  13 13 13 13 13 13 13 13  13 13 13 13 00 00 00 00  |................|
00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
```

Unfortunately I can't seem to mount them on my MiSTer. Normally when I pick a disk image that image name is shown on the menu, but no matter what I pick, it always shows `Mount A: *.DSK`, and I can't get the system to boot.

Do I need to convert them to another format in order to use them with this core?

**Comment by steddyman** [comment]
Hi

Are you trying to mount them in drive A or drive B and are you in 8256/8512 mode?  Unfortunately the models are not well named.  The 8256/8512 is badly named, and really means a PCW with a Dot Matrix printer ROM.   A real PCW 8256 comes with a CF2 drive in A and an optional CF2DD in B.  So the double stepping logic will only apply to drive B.  I have read Extended disk formats without a problem in drive B in that mode.

The 9256/9512 mode is a PCW with a Daisywheel Printer ROM.  That supports MF2DD (3.5 720k disks) in both drive A and drive B, but of course requires different CPM boot disks in that new format.

**Comment by thejpster** [comment]
I have a real 9512, but I was trying to boot a CF2 8256 disk image from my collection using drive A:. I will try a CF2DD in drive B: and report back.

The 9256 and 9512+ have 3.5" drives, whilst the 9512 has a single CF2HD. Which of those is emulated in "9256/9512" mode?

**Comment by steddyman** [comment]
Really, there should be three models in the core to accurately reflect the hardware which is:

8256 / 8512: Dot-matrix PCWs with (usually) 180k boot drives. Boot sector checksum is 0FFh.
9256 / 10: Dot-matrix PCWs with 720k boot drives. Boot sector checksum is 0FFh.
9512 / 9512+: Daisywheel PCWs with 720k boot drives. Boot sector checksum is 01h.

In the core, the 8256/8512 mode is 1 x CF2 (180k) and 1 x CF2DD (720k).  The 9256/9512 mode is 2 x MF2DD (720k).

**Comment by thejpster** [comment]
I agree, provided the image files for an 3.5" 720K MF2DD are the same as the image files for a 3" 720K CF2DD. I know they are both 80 track, 9 sector/track, 512 byte/sector formats, but I don't know if there are any other differences.

**Comment by steddyman** [comment]
Yes, from a geometry and format perspective they are identical.  It is only the physical size that changes.
